src/common.c

```diff
@@ -64,7 +65,20 @@ int fi_poll_fd(int fd, int timeout)
 
 	fds.fd = fd;
 	fds.events = POLLIN;
-	ret = poll(&fds, 1, timeout);
+
+	// to set the signal mask
+        sigset_t orgmask, sigmask;
+
+         /* Block SIGUSR1, SIGUSR2, and all realtime signals from SIGRTMIN to SIGRTMAX */
+        sigemptyset(&sigmask);
+        sigaddset(&sigmask, SIGUSR1);
+        sigaddset(&sigmask, SIGUSR2);
+	for (int i = SIGRTMIN; i <= SIGRTMAX; i++)
+	        sigaddset(&sigmask, i);
+        pthread_sigmask(SIG_SETMASK, &sigmask, &orgmask);
+        ret = poll(&fds, 1, timeout);
+        pthread_sigmask(SIG_SETMASK, &orgmask, NULL);
```

Comment:
nit - Indentation is off above - please use tabs in place of spaces.

Also I thought this code structure was racy, and why ppoll was added.